### PR TITLE
[fixup] fixed promises/events to work with ring buffer change

### PR DESCRIPTION
**Pull request by jprestwo**
Signed-off-by: James Prestwood <james.prestwood@intel.com>

### CODE REVIEW

File: src/zjs_promise.c [hunk 1]
```diff
@@ -113,6 +116,7 @@ void zjs_make_promise(jerry_value_t obj, zjs_post_promise_func post,
     new->post = post;
     new->then_set = 0;
     new->catch_set = 0;
+    new->then = jerry_acquire_value(obj);
```
Comment (grgustaf) [inline src/zjs_promise.c]: It seems like this acquired object will not be released, because "then_set" is checked and this gets overwritten later.

**Comment by grgustaf** [comment]
LGTM, will merge this after Travis passes. Short-circuiting Brady Bill waiting period because this came from the iotivity-constrained patch and holds up two other PRs.

File: src/zjs_promise.c [hunk 2]
```diff
@@ -124,46 +128,64 @@ void zjs_make_promise(jerry_value_t obj, zjs_post_promise_func post,
 
 void zjs_fulfill_promise(jerry_value_t obj, jerry_value_t argv[], uint32_t argc)
 {
-    struct promise* handle;
+    struct promise* handle = NULL;
     jerry_value_t promise_obj = zjs_get_property(obj, "promise");
```
Comment (jimmy-huang) [inline src/zjs_promise.c]: I think you have to release the promise_obj when you exit this func

File: src/zjs_promise.c [hunk 3]
```diff
@@ -124,46 +128,64 @@ void zjs_make_promise(jerry_value_t obj, zjs_post_promise_func post,
...
+                                                handle,
+                                                post_promise);
 
-    zjs_signal_callback(handle->then_id, argv, argc * sizeof(jerry_value_t));
+        zjs_signal_callback(handle->then_id, argv, argc * sizeof(jerry_value_t));
 
-    DBG_PRINT("fulfilling promise, obj=%lu, then_id=%lu, argv=%p, nargs=%lu\n",
-              obj, handle->then_id, argv, argc);
+        DBG_PRINT("fulfilling promise, obj=%lu, then_id=%lu, argv=%p, nargs=%lu\n",
+                obj, handle->then_id, argv, argc);
```
Comment (jimmy-huang) [inline src/zjs_promise.c]: align indent here

File: src/zjs_promise.c [hunk 4]
```diff
@@ -124,46 +128,64 @@ void zjs_make_promise(jerry_value_t obj, zjs_post_promise_func post,
...
+    } else {
+        ERR_PRINT("native handle not found\n");
+    }
 }
 
 void zjs_reject_promise(jerry_value_t obj, jerry_value_t argv[], uint32_t argc)
 {
-    struct promise* handle;
+    struct promise* handle = NULL;
     jerry_value_t promise_obj = zjs_get_property(obj, "promise");
```
Comment (jimmy-huang) [inline src/zjs_promise.c]: again, you'll have to release the promise_obj I think

File: src/zjs_promise.c (hunk 3 above)
Comment (jprestwo) [inline src/zjs_promise.c]: how should I indent this? what is our policy on this?

Comment (grgustaf) [inline src/zjs_promise.c]: Minor, but either obj should line up with " or be 4-chars in, I think we allow either.

**Comment by grgustaf** [comment]
+1, passing now.

**Pull request merged**